core/src/main/java/org/apache/calcite/rel/core/Sample.java

```diff
@@ -58,12 +58,24 @@ public Sample(RelInput input) {
   private static RelOptSamplingParameters getSamplingParameters(
       RelInput input) {
     String mode = input.getString("mode");
-    float percentage = input.getFloat("rate");
+    Object percentage = input.get("rate");
+    Object numberOfRows = input.get("rows");
     Object repeatableSeed = input.get("repeatableSeed");
     boolean repeatable = repeatableSeed instanceof Number;
-    return new RelOptSamplingParameters(
-        "bernoulli".equals(mode), percentage, repeatable,
-        repeatable && repeatableSeed != null ? ((Number) repeatableSeed).intValue() : 0);
+
+    if (percentage instanceof Number) {
```

Comment:
I'm guessing this is because percentage and number of rows can be null. At the same time, this is an anti-pattern to use `instanceof` for this circumstance. Can we have another way to determine whether to use the rate or rows from the input?